### ISSUE TITLE
Limitations of Kanata implementation 

### ISSUE THREAD

**Issue by argenkiwi** [dreamsofcode-io/home-row-mods#9]
I have seen some common complaints in the comments regarding the implementation of home row modifiers using Kanata:
- Error prone
- Laggy 
- Can't repeat keys when held

All of those can be addressed using a more elaborate approach (like [this one](https://github.com/jtroo/kanata/discussions/1455#discussion-7766783)). 

It is great you are raising awareness of the existence of software like Kanata, but I fear user's will get the wrong impression and think they are not as powerful as they actually are.

**Comment by elliottminns** [comment]
Thanks for raising this and providing the more elaborate approach.

So as not deviate too much from the video, I suggest we add this as an alternative option in the kanata directory and instead merge #7 in order to solve any issues that viewers might be having.

**Comment by argenkiwi** [comment]
Thanks @elliottminns. Sounds good to me. As I mentioned in the pull request, that fixes an issue that affects Linux users specifically and may add unnecessary complexity to Windows and MacOS users. 

It may be prudent to check whether using one of the alternatives to `tap-hold` could bypass that specific issue completely. 

In any case, thanks for looking into this and looking forward to more of your content in 2025.

**Comment by camilolunacom** [comment]
@argenkiwi @elliottminns  I see you mentioned "Can't repeat keys when held" which I use a lot to navigate in vim. Is there an easy fix for this or would you suggest looking at the switch linked on the issue description?

**Comment by argenkiwi** [comment]
> [@argenkiwi](https://github.com/argenkiwi) [@elliottminns](https://github.com/elliottminns) I see you mentioned "Can't repeat keys when held" which I use a lot to navigate in vim. Is there an easy fix for this or would you suggest looking at the switch linked on the issue description?

Absolutely, it is a matter of using `tap-hold-realease-timeout`. You can see my implementation [here](https://github.com/argenkiwi/kenkyo/blob/main/kanata/components/main.kbd).

The switch part is to skip the HRMs while typing at speed, which reduces lag and misfires.

**Comment by camilolunacom** [comment]
@argenkiwi thank you so much! I tried the switch but didn't like it, so I reverted and replaced `tap-hold` with `tap-hold-release-timeout`(adding the additional parameter) and I'm loving it so far!

**Comment by argenkiwi** [comment]
@camilolunacom Great to hear! The tradeoff of using switch is that using home row mods in the middle of a sentence gets a bit tricky, as you need to make a small pause so the modifier does not get skipped. But if you are used to the small input lag that plain `tap-hold-*` requires, you can absolutely live without using switch.

**Comment by sgloutnikov** [comment]
With the original implementation on in this repo on macOS I notice quite significant lag when typing, almost to the point that it was a deal breaker for me. The `.kbd` posted above by @argenkiwi improves things a lot! Very minimal lag almost unnoticeable, and navigating the cursor with `ctrl-f` or `ctrl-b` for example is near identical to using the stock experience but still noticeable laggy for me. Thanks for posting, I'll look into testing some more.
